src/main/java/com/maxmind/minfraud/request/Email.java

```diff
@@ -46,7 +49,34 @@ private Email(Email.Builder builder) {
             if (this.domain == null) {
                 this.domain = address.substring(address.indexOf('@') + 1);
             }
-            this.addressMd5 = DigestUtils.md5Hex(address);
+            this.address = address;
+            return this;
+        }
+
+        /**
+         * Set the email address and domain fields for the request. If
+         * you set the email address from this method, you do <em>not</em>
+         * need to set the domain separately. The domain will be set to
+         * the domain of the email address and the address field will be
+         * set to the MD5 of the email address passed.
+         *
+         * The MD5 of the email address will be sent. If you want to send the
+         * raw email address, use {@link #address(String)} instead.
+         *
+         * @param address The valid email address used in the transaction.
+         * @return The builder object.
+         * @throws IllegalArgumentException when address is not a valid email
+         *                                  address.
+         */
+        public Email.Builder addressMd5(String address) {
```

Comment:
I wonder if people who don't read the documentation closely will be confused by this name and think you have to pass an MD5 to the method. This might particularly be an issue for people familiar with the old service. Validation does help here though so maybe it isn't an issue.

Some people will also likely call both `.address()` and `.addressMd5()` on the same builder.

Another option that I had been considering is rather than adding a separate setter method on the builder, we could instead have a method to enable encoding of the address as an MD5. So you might do something like:

```
        new Email.Builder()
            .address("fraud@ster.com")
            .hashAddress() // there is probably a better name for this
            .domain("ster.com")
            .build()
```

Reply:
The point about potentially calling both of them is good. `hashAddress()` is pretty clear. `hash()` might be nice too but could be misleading.